redaxo/src/core/lib/sql/table.php

```diff
@@ -141,56 +172,177 @@ public function removeColumn($name)
         return $this;
     }
 
+    /**
+     * @return string[] Column names
+     */
+    public function getPrimaryKey()
+    {
+        return $this->primaryKey;
+    }
+
+    /**
+     * @param string|string[] $columns Column name(s)
+     *
+     * @return $this
+     */
+    public function setPrimaryKey($columns)
+    {
+        $columns = (array) $columns;
+
+        if ($this->primaryKey === $columns) {
+            return $this;
+        }
+
+        $this->primaryKey = (array) $columns;
+        $this->primaryKeyModified = true;
+
+        return $this;
+    }
+
+    /**
+     * Ensures that the table exists with the given definition.
+     */
+    public function ensure()
+    {
+        if ($this->new) {
+            $this->create();
+
+            return;
+        }
+
+        $this->alter();
+    }
+
+    /**
+     * Drops the table if it exists.
+     */
+    public function drop()
+    {
+        if (!$this->new) {
+            $this->sql->setQuery(sprintf('DROP TABLE %s', $this->sql->escapeIdentifier($this->name)));
+        }
+
+        $this->new = true;
+        $this->columnsExisting = [];
+        $this->primaryKeyModified = [] !== $this->primaryKey;
+    }
+
+    /**
+     * Creates the table.
+     *
+     * @throws rex_exception
+     */
+    public function create()
+    {
+        if (!$this->new) {
+            throw new rex_exception(sprintf('Table "%s" already exists.', $this->name));
+        }
+        if (!$this->columns) {
+            throw new rex_exception('A table must have at least one column.');
+        }
+
+        $parts = [];
+
+        foreach ($this->columns as $column) {
+            $parts[] = $this->getColumnDefinition($column);
+        }
+
+        if ($this->primaryKey) {
+            $parts[] = 'PRIMARY KEY '.$this->getKeyColumnsDefintion($this->primaryKey);
+        }
+
+        $query = 'CREATE TABLE '.$this->sql->escapeIdentifier($this->name)." (\n    ";
+        $query .= implode(",\n    ", $parts);
+        $query .= "\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;";
```

Comment:
jo passt, aktuell ist es ja in redaxo überall gleich. sobald es jmd flexibel brauch kann man es ändern.